docker/Dockerfile

```diff
@@ -1,31 +1,42 @@
 FROM centos:5.11
 MAINTAINER The NumPy Project
 
-# Base image (~ reproduce the packer base image)
-RUN yum -y install gcc bzip2 make kernel-devel-`uname -r`
-RUN yum -y install gcc-c++
-RUN yum -y erase wireless-tools gtk2 libX11 hicolor-icon-theme avahi freetype bitstream-vera-fonts
-RUN yum -y install curl wget
-RUN yum -y clean all
-
 # EPEL support
-RUN wget https://dl.fedoraproject.org/pub/epel/5/x86_64/epel-release-5-4.noarch.rpm
-RUN rpm -Uvh --replacepkgs epel-release-5*.rpm
-RUN rm -f epel-release-5*.rpm
+ADD https://dl.fedoraproject.org/pub/epel/5/x86_64/epel-release-5-4.noarch.rpm /
+# Dev toolset (for LLVM and other projects requiring C++11 support)
+ADD http://people.centos.org/tru/devtools-2/devtools-2.repo /etc/yum.repos.d/
+RUN rpm -Uvh --replacepkgs epel-release-5*.rpm && rm -f epel-release-5*.rpm
 
-# Install python26
-RUN yum install -y python26 python26-devel
-RUN curl -L https://bootstrap.pypa.io/get-pip.py --output /tmp/get-pip.py
-RUN /usr/bin/python2.6 /tmp/get-pip.py
 
-# Dev toolset (for LLVM and other projects requiring C++11 support)
-ADD devtoolset-repo.repo /etc/yum.repos.d/devtoolset-repo.repo
-RUN yum -y install devtoolset-2-binutils devtoolset-2-gcc devtoolset-2-gcc-c++
+# Development tools and libraries
+RUN yum -y install curl wget bzip2 make git patch unzip bison yasm \
+                   gcc-c++ gcc  kernel-devel-`uname -r` \
+                   devtoolset-2-binutils devtoolset-2-gcc devtoolset-2-gcc-c++ \
+                   gcc-gfortran gcc44 gcc44-c++ gcc44-gfortran \
+                   mesa-libGL-devel mesa-libGLU-devel libXt-devel cups-devel \
+                   libXext libSM libXrender
+
+RUN yum -y erase wireless-tools gtk2 libX11 hicolor-icon-theme \
+                 avahi freetype bitstream-vera-fonts && \
+    yum -y clean all
+
+
+# Python compile-time dependencies
+RUN yum install -y zlib-devel bzip2-devel openssl-devel ncurses-devel \
+                   sqlite-devel readline-devel tk-devel gdbm-devel db4-devel \
+                   libpcap-devel xz-devel
+# Should the headers from these -devel packages be deleted after we finish
+# building Python?
```

Comment:
do you know if all of those are actually necessary (i just copied that from somewhere on the web). not sure if python needs xz, gdbm, db4...